Sources/StreamChat/Database/DTOs/ChannelDTO.swift

```diff
@@ -351,10 +351,14 @@ extension ChannelDTO {
         // We can't pass bools directly to NSPredicate so we have to use integers
         let isHidden = NSPredicate(format: "isHidden == %i", query.filter.hiddenFilterValue == true ? 1 : 0)
 
-        let subpredicates = [
+        var subpredicates: [NSPredicate] = [
             matchingQuery, notDeleted, isHidden
         ]
 
+        if StreamRuntimeCheck.isChannelLocalFilteringEnabled, let filterPredicate = query.filter.predicate {
```

Comment:
If we use the `ChatClientConfig`, we can pass this in the `channelListFetchRequest()` method. There's no reason for this to be static. CC @polqf @martinmitrevski 

We currently have config flags spread in multiple places: `Components`, `ChatClientConfig` and `StreamRuntimeCheck`. The `Components` should be used for UI (Although it deserves a different struct as well, right now it handles too much), and the `ChatClientConfig` should be used for LLC configs. The `StreamRuntimeCheck` should be unnecessary, I actually thought this was a private API 🤔